connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/Health.java

```diff
@@ -102,20 +133,51 @@ public static Health down(Throwable ex) {
     return Health.status(Status.DOWN).detail("error", error);
   }
 
+  public static Health down(Throwable ex, String title) {
+    HealthError healthError =
+        new HealthError(Severity.ERROR, ex.getMessage(), getStackTrace(ex), title);
+    return Health.status(Status.DOWN).error(healthError);
+  }
+
+  public Health merge(Health newHealth) {
+    Map<String, Object> mergedDetails =
+        this.getDetails() != null ? this.getDetails() : new HashMap<>();
+    if (newHealth.getDetails() != null && !newHealth.getDetails().isEmpty()) {
+      mergedDetails.putAll(newHealth.getDetails());
+    }
+    return Health.status(newHealth.getStatus()).details(mergedDetails).error(newHealth.getError());
+  }
```

Comment:
Right now we have only 1 single Heath instance per inbound connector instance. If we would like to add a new details for example, than we would like to keep the original details(e.g.: we already had the path, but we also want to add the authentication type just to show it there), and only append the new one instead of overriding it completely. Hence the name of the method `merge`.
Better example is when you wanna change the status attribute of the Health object. We always have to show the details(e.g. path for webhook), so we cannot override that, so merging the information is our only option.

Reply:
@markfarkas-camunda So should we maybe rather have a merge for properties only?

Reply:
@markfarkas-camunda Sorry, I meant then maybe just allow users to pass details instead of a whole health object.

Reply:
@sbuettner but somehow they need to change the status, ~and error as well~(this is irrelevant if we will use the details), and they might want to change those at the same time. Do you mean we should have different methods for changing different attributes? So one method for merging details, one method for changing status and maybe one for creating health object? But in this case how could we prevent them to accidentally override the Health object, and with that deleting the details? Since we should handle the "error" detail specially we would also have to have at least one method for changing that.

Reply:
What kind of details should we keep and we cant easily created again?